lock-api/src/test/java/com/palantir/lock/client/LeasingTimelockClientTest.java

```diff
@@ -0,0 +1,222 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.lock.client;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+import java.time.Duration;
+import java.util.Set;
+import java.util.UUID;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import com.google.common.collect.ImmutableSet;
+import com.palantir.common.time.NanoTime;
+import com.palantir.lock.v2.IdentifiedTime;
+import com.palantir.lock.v2.LeasableLockResponse;
+import com.palantir.lock.v2.LeasableRefreshLockResponse;
+import com.palantir.lock.v2.LeasableStartIdentifiedAtlasDbTransactionResponse;
+import com.palantir.lock.v2.Lease;
+import com.palantir.lock.v2.LockImmutableTimestampResponse;
+import com.palantir.lock.v2.LockRequest;
+import com.palantir.lock.v2.LockResponse;
+import com.palantir.lock.v2.LockToken;
+import com.palantir.lock.v2.StartIdentifiedAtlasDbTransactionRequest;
+import com.palantir.lock.v2.StartIdentifiedAtlasDbTransactionResponse;
+import com.palantir.lock.v2.TimelockService;
+import com.palantir.lock.v2.TimestampAndPartition;
+
+public class LeasingTimelockClientTest {
+    private TimelockRpcClient timelockService = mock(TimelockRpcClient.class);
+    private TimelockService timelockClient;
+
+    private static final UUID LEADER_ID = UUID.randomUUID();
+
+    private static final LockToken LOCK_TOKEN = LockToken.of(UUID.randomUUID());
+    private static final LockResponse LOCK_RESPONSE = LockResponse.successful(LOCK_TOKEN);
+
+    @Before
+    public void setUp() {
+        when(timelockService.getLeaderTime()).thenReturn(IdentifiedTime.of(UUID.randomUUID(), NanoTime.now()));
+        timelockClient = LeasingTimelockClient.create(timelockService);
+    }
+
+    @Test
+    public void delegatesLockRequest() {
+        LockRequest lockRequest = mock(LockRequest.class);
+        when(timelockService.leasableLock(lockRequest)).thenReturn(
+                LeasableLockResponse.of(LOCK_RESPONSE, getLease()));
+
+        timelockClient.lock(lockRequest);
+
+        verify(timelockService).leasableLock(lockRequest);
+    }
+
+    @Test
+    public void lockResponeHasCorrectLeasedLock() {
+        LockRequest lockRequest = mock(LockRequest.class);
+        Lease lease = getLease();
+        when(timelockService.leasableLock(lockRequest)).thenReturn(
+                LeasableLockResponse.of(LOCK_RESPONSE, lease));
+
+        LockResponse clientResponse = timelockClient.lock(lockRequest);
+        LeasedLockToken leasedLockToken = (LeasedLockToken) clientResponse.getToken();
+        assertEquals(LOCK_TOKEN, leasedLockToken.serverToken());
```

Comment:
you should never use assertEquals, it's too error-prone (people flip expected and actual all the time). Prefer using assertThat() every time.

Reply:
(so that's a comment about every assertEquals in these files)